src/Evolution/Systems/CurvedScalarWave/Worldtube/SelfForce.cpp

```diff
@@ -0,0 +1,64 @@
+
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/SelfForce.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace CurvedScalarWave::Worldtube {
+
+template <size_t Dim>
+void self_force_acceleration(
+    gsl::not_null<tnsr::I<double, Dim>*> self_force_acc,
+    const Scalar<double>& dt_psi_monopole,
+    const tnsr::i<double, Dim>& psi_dipole,
+    const tnsr::I<double, Dim>& particle_velocity, const double particle_charge,
+    const double particle_mass, const tnsr::AA<double, Dim>& inverse_metric,
+    const Scalar<double>& dilation_factor) {
+  const double u0_squared = square(get(dilation_factor));
+  for (size_t i = 0; i < Dim; ++i) {
+    self_force_acc->get(i) =
+        (inverse_metric.get(i + 1, 0) -
+         particle_velocity.get(i) * inverse_metric.get(0, 0)) *
+        get(dt_psi_monopole) * particle_charge / particle_mass / u0_squared;
```

Comment:
[optional] Factor out the computation of `particle_charge / particle_mass / u0_squared` so it isn't recomputed. (useful also for the loop below)